### PR TITLE
Precompiled contract for enode checking

### PR DESCRIPTION
**Pull request by JekaMas**
shown the idea of #487

### CODE REVIEW

**Comment by yazzaoui** [comment]
Yes looking good with a separate library contract with the assembly call.

File: contracts/autonity/contract/contracts/Precompiled.sol [hunk 1]
```diff
@@ -0,0 +1,13 @@
+pragma solidity ^0.5.0;
+
+library Precompiled {
+    function enodeCheck(string enode) internal pure returns (uint[2] memory p) {
+        assembly {
+            if iszero(staticcall(gas, 0xff, input, 0x80, p, 0x40)) {
```
Comment (piersy) [inline contracts/autonity/contract/contracts/Precompiled.sol]: It would be nice to have a comment here explaining exactly what this line is doing

Comment (JekaMas) [reply contracts/autonity/contract/contracts/Precompiled.sol]: I can provide a link to the doc. Shall we double the documentation in the comments? May not really.

Comment (piersy) [reply contracts/autonity/contract/contracts/Precompiled.sol:6]: @JekaMas  A link would be good.

 I'm not sure what you mean about doubling the documentation in the comments?

Comment (JekaMas) [reply contracts/autonity/contract/contracts/Precompiled.sol:6]: A bit strange to provide links to solidity docs in solidity code. Shall we do it?

Comment (piersy) [reply contracts/autonity/contract/contracts/Precompiled.sol:6]: @JekaMas I am confused, you suggested the link. 

I think it is just hard to parse when reading, potentially a comment could make that much simpler.

I don't understand how we came up with the contract address of `0xff` or why it returns `uint[2]` or why we are limiting the input to a length of 128 when enode strings are longer. Maybe this is just really obvious to everyone else, but not to me.

Comment (JekaMas) [reply contracts/autonity/contract/contracts/Precompiled.sol:6]: Unfortunately, it's all about solidity and evm they are complex and complicated. And it's the only way to call precomputed contract (you can read about it with examples in the link above).

Comment (piersy) [reply contracts/autonity/contract/contracts/Precompiled.sol:6]: @JekaMas Which link?

**Comment by JekaMas** [comment]
It's mentioned in the issue https://medium.com/@rbkhmrcr/precompiles-solidity-e5d29bd428c4

**Comment by piersy** [comment]
@JekaMas Thanks, I will take a look when I get a chance.

I think it could be nice to include that link in a comment, just so that for people unfamiliar with precompiled contracts in solidity they have the guide that, I guess you followed, right there.

**Comment by JekaMas** [comment]
If we include any links to the comments to the code then we have to support(check that they are still available) them somehow. It's not obvious to me how can we do that.

**Comment by piersy** [comment]
@JekaMas I agree it would be nice if we could have a way of checking the links validity automatically.

But even if the link went dead it can be useful while it is still alive, eventually someone will notice the link is not working  and we can update it.

Once the link goes dead it's no worse than the current state of not having any link at all.

**Comment by JekaMas** [comment]
Can't agree. Shall we provide links on how golang works or evm?

File: contracts/autonity/contract/contracts/Precompiled.sol (hunk 1 above)
Comment (JekaMas) [reply contracts/autonity/contract/contracts/Precompiled.sol]: The link is added to the code

File: .github/workflows/go.yml [hunk 2]
```diff
@@ -12,7 +12,7 @@ jobs:
         with:
           path: /home/runner/work/autonity/autonity/
           key: ${{ runner.os }}-aut-${{ env.cache-name }}-${{ github.sha }}
-          restore-keys: ${{ runner.os }}-aut-${{ env.cache-name }}-
+          restore-keys: ${{ runner.os }}-aut-${{ env.cache-name }}-${{ github.sha }}
```
Comment (piersy) [inline .github/workflows/go.yml:15]: I don't think, having a restore key that is the same as the key provides any value. Restore keys are meant as a backup in case the cache key cannot be found

File: .github/workflows/develop.yml [hunk 3]
```diff
@@ -1,5 +1,8 @@
 name: develop
-on: [push]
+on:
+  release:
```
Comment (piersy) [inline .github/workflows/develop.yml:3]: Just wondering why you changed this. It seems unrelated to this PR, but also won't this stop us getting feedback on whether TestTendermintTC7 untill it is too late (I.E the release has already been made)?

File: .github/workflows/go.yml (hunk 2 above)
Comment (JekaMas) [reply .github/workflows/go.yml:15]: Interesting, I've just implemented your idea about it. Why does it now not so good?

File: .github/workflows/develop.yml (hunk 3 above)
Comment (JekaMas) [reply .github/workflows/develop.yml:3]: I think this is the cause of flaky tests. After updating branch with recent develop, CI runs both workflows, for usual PR and for release. This took > 40m. However, using this flag everything works as expected.
So we don't have flaky tests, but we had too long CI by mistake @yazzaoui

File: .github/workflows/go.yml (hunk 2 above)
Comment (piersy) [reply .github/workflows/go.yml:15]: Hey, I'm not sure what you mean by

>Why does it now not so good?
